engine/crates/engine-v2/src/sources/graphql/deserialize/entities.rs

```diff
@@ -73,6 +76,7 @@ enum EntitiesKey {
 struct EntitiesSeed<'resp, 'parent> {
     response_part: &'parent SubgraphResponseRefMut<'resp>,
     plan: PlanWalker<'resp>,
+    cache_entries: Option<std::slice::Iter<'parent, CachedEntity>>,
```

Comment:
If you end up with an iterator in all cases, I would have made it generic over the iterator and provided an empty iterator in case nothing was cached, rather than an operation of an iterator.